kotsadm/web/src/components/apps/AppSnapshotSchedule.jsx

```diff
@@ -164,28 +164,49 @@ class AppSnapshotSchedule extends Component {
 
   saveSnapshotConfig = () => {
     this.setState({ updatingSchedule: true });
-    this.props.saveSnapshotConfig(
-      this.props.app.id,
-      this.state.retentionInput,
-      this.state.selectedRetentionUnit?.value,
-      this.state.frequency,
-      this.state.autoEnabled,
-    ).then(() => {
-      this.setState({ updatingSchedule: false, updateConfirm: true });
-      setTimeout(() => {
-        this.setState({ updateConfirm: false })
-      }, 3000);
+    const body = {
+      appId: this.props.app.id,
+      inputValue: this.state.retentionInput,
+      inputTimeUnit: this.state.selectedRetentionUnit?.value,
+      schedule: this.state.frequency,
+      autoEnabled: this.state.autoEnabled,
+    };
+    fetch(`${window.env.API_ENDPOINT}/app/${this.props.app.slug}/snapshot/config`, {
+      headers: {
+        "Authorization": Utilities.getToken(),
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+      },
+      method: "PUT",
+      body: JSON.stringify(body),
     })
-    .catch(err => {
-      console.log(err);
-      err.graphQLErrors.map(({ msg }) => {
+      .then(async (res) => {
+        const data = await res.json();
+        if (!res.ok || !data.success) {
+          if (res.status === 401) {
+            Utilities.logoutUser();
+            return;
+          }
+          this.setState({
+            message: data.error || "Failed to save snapshot config",
+            messageType: "error",
+            updatingSchedule: false,
+          })
+          return
+        }
+        this.setState({ updatingSchedule: false, updateConfirm: true });
+        setTimeout(() => {
+          this.setState({ updateConfirm: false })
+        }, 3000);
+      })
+      .catch((err) => {
+        console.log(err);
         this.setState({
-          message: msg,
+          message: "Failed to connect to API",
```

Comment:
you can check for `err` and use `err.message` and if it does not exists, use `"Failed to connect to API"`. something like
```suggestion
          message: err ? err.message : "Failed to connect to API",
```